extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -36,6 +36,34 @@ public static boolean isConfiguredWithWildcard(Optional<List<String>> optionalLi
         return list.isEmpty() || (list.size() == 1 && "*".equals(list.get(0)));
     }
 
+    /**
+     * If any regular expression origins are configured, try to match on them.
+     * Regular expressions must begin and end with '/'
+     * 
+     * @param allowedOrigins the configured oeigins.
+     * @param origin the specified origin
+     * @return true if any configured regular expressions match the specified origin, false otherwise
+     */
+    public static boolean isOriginAllowedByRegex(Optional<List<String>> allowedOrigins, String origin) {
+        if (allowedOrigins == null || !allowedOrigins.isPresent()) {
+            return false;
+        }
+
+        // extract configured origins and find any Regular Expressions
+        List<Pattern> allowOriginsRegex = new ArrayList<>();
```

Comment:
@ekasumov IMHO it should be precalculated during the object construction to avoid compiling the patterns on every request; also my understanding it would be recommended to use the lambda code in that case

Reply:
@sberyozkin would you recommend putting that in the CORSConfig class, or calculating it when CORSFilter is constructed?

Reply:
@ekasumov interesting, did you mean have `Pattern` directly in `CORSConfig` ? +1 - it would require a converter - but it would potentially of interest elsewhere in Quarkus as well - so please add this converter to the core module - if it is not already there. also CC @radcortez

Reply:
Or may be keep this converter in the vertx http for now - we can decide before merging what to do with it

Reply:
@sberyozkin just pushed the changes. The regex parsing and compilation has been moved to the constructor so that it's not performed on every request. Good call.

Reply:
I don't think we have a Pattern Converter. Feel free to add one.